app.go

```diff
@@ -81,9 +83,21 @@ func (f optionFunc) apply(app *App) { f(app) }
 //
 // See the documentation for go.uber.org/dig for further details.
 func Provide(constructors ...interface{}) Option {
-	return optionFunc(func(app *App) {
-		app.provides = append(app.provides, constructors...)
-	})
+	return provideOption(constructors)
+}
+
+type provideOption []interface{}
+
+func (po provideOption) apply(app *App) {
+	app.provides = append(app.provides, po...)
```

Comment:
Let's keep this PR as-is. We're already trying to figure out how to run invokes in `New` in the PR you linked, which will solve the error-detection problem one way or another.